File: flink-table-1.7.1/src/main/scala/scala/WordCountSQL.scala
package scala

import org.apache.flink.api.scala.ExecutionEnvironment
import org.apache.flink.api.scala._
import org.apache.flink.table.api.TableEnvironment
import org.apache.flink.table.api.scala._

import scala.util.WC

object WordCountSQL {
  def main(args: Array[String]): Unit = {
    //Environment
    val env = ExecutionEnvironment.getExecutionEnvironment
    val tableEnv = TableEnvironment.getTableEnvironment(env)
    //Source
    val input = env.fromElements(WC("hello", 1), WC("hello", 1), WC("ciao", 1))
    //Table
    tableEnv.registerDataSet("WordCount", input, 'word, 'frequency)
    //SQL
    val table = tableEnv.sqlQuery("SELECT word, SUM(frequency) FROM WordCount GROUP BY word")
    //DataSet
    val result = table.toDataSet[WC]
    //Sink
    result.print()
  }
}

File: flink-table-1.7.1/src/main/java/WordCountSQL.java
import org.apache.flink.api.java.DataSet;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.table.api.Table;
import org.apache.flink.table.api.TableEnvironment;
import org.apache.flink.table.api.java.BatchTableEnvironment;
import util.WC;

public class WordCountSQL {
    public static void main(String[] args) throws Exception {
        //Environment
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
        BatchTableEnvironment tableEnv = TableEnvironment.getTableEnvironment(env);
        //Source
        DataSet<WC> input = env.fromElements(
                new WC("Hello", 1),
                new WC("Ciao", 1),
                new WC("Hello", 1));
        //Table
        tableEnv.registerDataSet("WordCount", input, "word, frequency");
        //SQL
        Table table = tableEnv.sqlQuery("SELECT word, SUM(frequency) as frequency FROM WordCount GROUP BY word ORDER BY word");
        //DataSet
        DataSet<WC> result = tableEnv.toDataSet(table, WC.class);
        //Sink
        result.print();
    }
}
